security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -567,12 +577,115 @@ func (sc *SecretCache) generateGatewaySecret(token string, connKey ConnKey, t ti
 	}, nil
 }
 
+// If there is an existing root certificate under a well known path, return true.
+// Otherwise, return false.
+func (sc *SecretCache) rootCertificateExists() bool {
+	b, err := ioutil.ReadFile(existingRootCertFile)
+	if err != nil || len(b) == 0 {
+		return false
+	}
+	return true
+}
+
+// If there is an existing private key and certificate under a well known path, return true.
+// Otherwise, return false.
+func (sc *SecretCache) keyCertificateExists() bool {
+	b, err := ioutil.ReadFile(existingCertChainFile)
+	if err != nil || len(b) == 0 {
+		return false
+	}
+	b, err = ioutil.ReadFile(existingRootCertFile)
```

Comment:
I think you can delete this - probably not a big deal, since the interesting case is having root cert but not keys. 
Not critical.